### PR TITLE
Refactor Navigation and Course Detail Screen Enhancements

### PR DESCRIPTION
**Pull request by nguyenminhhung2011**
| Status  | Type  | Env Vars Change |
| :---: | :---: | :---: |
| Ready | Feature/Refactor | No |

## Problem

- Complexity in handling multiple navigator instances within the app.
- Hardcoded navigation paths leading to limited flexibility.
- Lack of back navigation functionality in course detail screens.
- Inconsistent handling of course item click events.

## Solution

- Simplified navigation setup by using a single navigator instance.
- Changed navigation destinations to more generic ones for improved flexibility.
- Introduced back navigation feature in course detail screens.
- Updated course item click events to pass specific course IDs for better event handling.

## Test results

_Add your screenshots or recording about testing_

## Other changes

- Modified `BottomAppNavigation` to use `HorizontalPager` instead of `NavHost`.
- Updated `CourseDetail` object route to include course ID as a path parameter.
- Changed `BottomNavigationWrapper` function to handle both route and index on tab change.

## Deploy Notes

There are no new dependencies, scripts, or environment variables introduced with this PR.

### CODE REVIEW

File: navigation/src/main/java/com/example/navigation/bottomAppNavigation/BottomAppNavigation.kt [hunk 1]
```diff
@@ -23,47 +26,31 @@ import kotlinx.coroutines.flow.collectLatest
...
-                    navController.navigate(it.destination, builder = it.navOptions)
-                }
-
-                is Navigator.NavigationActions.Back -> {
-                    navController.popBackStack()
-                }
-            }
-        }
-    }
+    val state = rememberPagerState(initialPage = 0, pageCount = { bottomNavigationTabs.size })
```
Comment (nguyenminhhung2011) [inline navigation/src/main/java/com/example/navigation/bottomAppNavigation/BottomAppNavigation.kt:38]: Using `rememberPagerState` without providing `initialPage` and `pageCount` explicitly can lead to potential issues if the `bottomNavigationTabs` size changes dynamically.

```diff
-    val state = rememberPagerState(initialPage = 0, pageCount = { bottomNavigationTabs.size })
+    val state = rememberPagerState(initialPage = 0, pageCount = bottomNavigationTabs.size)
```

File: navigation/src/main/java/com/example/navigation/bottomAppNavigation/BottomAppNavigation.kt [hunk 2]
```diff
@@ -23,47 +26,31 @@ import kotlinx.coroutines.flow.collectLatest
...
-            ) {
-                composable(Destination.home.route) { homeScreen() }
-                composable(Destination.document.route) { documentScreen() }
-                composable(Destination.search.route) { searchScreen() }
+            HorizontalPager(state = state) { page ->
+                when (page) {
+                    0 -> homeScreen()
+                    1 -> documentScreen()
+                    2 -> searchScreen()
+                }
```
Comment (nguyenminhhung2011) [inline navigation/src/main/java/com/example/navigation/bottomAppNavigation/BottomAppNavigation.kt:48]: The use of hardcoded indices in the `when` block can lead to maintenance issues if the order of tabs changes or more tabs are added.

```diff
-                when (page) {
-                    0 -> homeScreen()
-                    1 -> documentScreen()
-                    2 -> searchScreen()
+                when (bottomNavigationTabs[page].route) {
+                    Destination.home.route -> homeScreen()
+                    Destination.document.route -> documentScreen()
+                    Destination.search.route -> searchScreen()
```

File: navigation/src/main/java/com/example/navigation/bottomAppNavigation/BottomAppNavigation.kt [hunk 3]
```diff
@@ -23,47 +26,31 @@ import kotlinx.coroutines.flow.collectLatest
...
             }
         },
-        onChangeTab = { destination ->
-            rememberDestination.value = destination
-            navController.navigate(destination)
+        onChangeTab = { route, pageIndex ->
+            rememberDestination.value = route
+            state.requestScrollToPage(pageIndex)
```
Comment (nguyenminhhung2011) [inline navigation/src/main/java/com/example/navigation/bottomAppNavigation/BottomAppNavigation.kt:53]: Directly using `state.requestScrollToPage(pageIndex)` may cause inconsistency issues. It's better to check if the requested page is different from the current page before making a call.

```diff
-            state.requestScrollToPage(pageIndex)
+            if (state.currentPage != pageIndex) {
+                state.requestScrollToPage(pageIndex)
+            }
```

**Review by hungmai2411 — APPROVED** [review]
review ok

**Pull request merged**